### PR TITLE
Added ccache tool.

### PR DESCRIPTION
**Pull request by fire**
Co-Authored-By: Daniel Moody <daniel.moody@mongodb.com>

Salvaging https://github.com/godotengine/godot/pull/52912.

### CODE REVIEW

**Comment by Calinou** [comment]
Out of curiosity, how does this compare with system-wide ccache integration (which can be performed by Linux distribution package managers by installing a wrapper)?

**Comment by fire** [comment]
I didn't see system level ccache enable itself in my testing in fedora or popos.

File: SConstruct [hunk 1]
```diff
@@ -290,6 +291,14 @@ if selected_platform in platform_opts:
 opts.Update(env_base)
 env_base["platform"] = selected_platform  # Must always be re-set after calling opts.Update().
 
+from SCons.Script.Main import _load_site_scons_dir
+
+_load_site_scons_dir(".", "misc/scons")
+
+ccache = Tool("ccache")
```
Comment (Faless) [inline SConstruct:298]:
```suggestion
ccache = Tool("ccache", toolpath=["misc/scons/site_tools"])
```

**Comment by fire** [comment]
We're still using this and it has a large performance speedup. Salvagable.

**Pull request closed without merging**